cmd/kpod/ps.go

```diff
@@ -0,0 +1,367 @@
+				}
+			}
+		} else {
+			containersOutput = containers
+		}
+
+		for _, ctr := range containersOutput {
+			ctrID = ctr.ID
+			runningFor = units.HumanDuration(time.Since(ctr.State.Created))
+			command = getCommand(ctr.ImageCreatedBy)
+			imageName = getImageName(ctr.FromImage, opts.noTrunc)
+			mounts = getMounts(ctr.Mounts, opts.noTrunc)
+			ports = getPorts(ctr.Config.ExposedPorts)
+			size = units.HumanSize(float64(ctr.SizeRootFs))
+
+			if ctr.State.Status == "stopped" {
+				status = "Exited (" + string(ctr.State.ExitCode) + ") " + runningFor + " ago"
+			} else if ctr.State.Status == "running" {
+				status = "Up " + runningFor + " ago"
+			} else {
+				status = "Created"
+			}
+
+			if !opts.noTrunc {
+				ctrID = ctr.ID[:idTruncLength]
+			}
+
+			params := psOutputParams{
+				ID:         ctrID,
+				Image:      imageName,
+				Command:    command,
+				CreatedAt:  ctr.State.Created,
+				RunningFor: time.Since(ctr.State.Created),
+				Status:     status,
+				Ports:      ports,
+				Size:       size,
+				Names:      ctr.Name,
+				Labels:     ctr.Labels,
+				Label:      "",
+				Mounts:     mounts,
+			}
+			output = append(output, params)
+		}
+	}
+	var out formats.Writer
+
+	if opts.format != "" {
+		switch opts.format {
+		case "json":
+			out = formats.JSONstruct{Output: toGenericPS(output)}
+		default:
+			out = formats.StdoutTemplate{Output: toGenericPS(output), Template: opts.format}
+		}
+	} else {
+		out = outStruct{output: output, opts: opts}
+	}
+
+	formats.Writer(out).Out()
+	return nil
+}
+
+// getCommand gets the actual command from the whole command
+func getCommand(cmd string) string {
+	start := strings.Index(cmd, "[")
+	if start == -1 {
+		return ""
+	}
+	start += len("[")
+	end := strings.Index(cmd, "]")
+	return cmd[start:end]
+}
+
+// getImageName shortens the image name when noTrunc is false
+func getImageName(img string, noTrunc bool) string {
+	var out string
+	if !noTrunc {
+		arr := strings.Split(img, "/")
+		if arr[0] == "docker.io" {
+			out = strings.Join(arr[2:], "/")
+		} else {
+			out = img
+		}
+	} else {
+		out = img
+	}
+	return out
+}
+
+// getLabels converts the labels to a string of the form "key=value, key2=value2"
+func getLabels(labels fields.Set) string {
+	var arr []string
+	if len(labels) > 0 {
+		for key, val := range labels {
+			temp := key + "=" + val
+			arr = append(arr, temp)
+		}
+		return strings.Join(arr, ",")
+	}
+	return ""
+}
+
+// getLabelVal gets the value of the key given for the label
+func getLabelVal(key string, labels fields.Set) string {
+	if val, ok := labels[key]; ok {
+		return val
+	}
+	return ""
+}
+
+// getMounts converts the volumes mounted to a string of the form "mount1, mount2"
+// it truncates it if noTrunc is false
+func getMounts(mounts []specs.Mount, noTrunc bool) string {
+	var arr []string
+	if len(mounts) > 0 {
+		for _, mount := range mounts {
+			if !noTrunc {
+				if strings.Contains(mount.Source, "/") {
```

Comment:
No reason to check, just split it and if it only has one arg then you know it had no /